### ISSUE TITLE
Specifying read types per file input

### ISSUE THREAD

**Issue by Neato-Nick** [MeHelmy/princess#9]
Is there a way to specify the -r option per read? My samples are a mix of pacbio and nanopore reads. I was guessing the SV and SNP genotyping would be more accurate if I called on the whole population at once, rather than separate princess runs for nanopore and pacbio.

Let me know if this is just not possible, or if merging results post-hoc would be just as good.

Thanks!

**Comment by MeHelmy** [comment]
Hi @Neato-Nick,
Thank you for reaching, Princess works on either PacBio or Nanopore but not a combination of both. Because each technology has its own error model. So, combining them while calling variants is not the best strategy.  Thus, I would suggest calling variants for each technology. Then, we can merge and compare the variants. After that, we identify the unique variant per technology and do enforcement calling for it to see if it is a variant or false-positive result. 

I have a question, do you have PacBio HiFi or CLR? What is your coverage? 
Best,
Medhat

**Comment by Neato-Nick** [comment]
Ah ok, that makes sense!

I have CLR and not HiFi, though the coverage is very high (150-320x). 1-2
SMRT cells off a sequel I a couple of years ago before HiFi preps were
common.

**Comment by MeHelmy** [comment]
That is perfect. So, you can use both CLR and ONT. If you care only for variants, you can use `princess variant` option, which will call SNVs, indels, and SVs. Later, you can merge and compare the result; take as an example  SVs, you can merge them using SURVIVOR. After that, you will identify the unique SVs for each technology and use the Sniffles enforcement method to confirm if it is a real variant or some error. For such high coverage, I suggest you change the field in `princess/config.yaml` file from:

```
# Structural Variant Parameters
###############################
sniffles_coverage: 2
```
To
```
# Structural Variant Parameters
###############################
sniffles_coverage: 10
```

Here we are saying that at least 10 reads needed to call an SV.

Best,
Medhat

**Comment by Neato-Nick** [comment]
Ah, that higher coverage setting will be great to make up for the noise in
CLR. Nanopore has like 40x coverage I think, I should've mentioned that
it's only pacbio I have that deep of coverage.

My plan was to use these long reads to check for structural variation but
also to come up with a reference panel of SNPs which I could then use with
to phase populations I've sequenced with Illumina. Read-based phasing with
long reads seems more trustworthy to me than phasing populations with an
unknown pedigree. With Princess calling SNPs and phasing it sounds like I
can avoid WhatsHap as an extra step.

**Comment by MeHelmy** [comment]
Right, So If you want to phase also SNVs, then you are on the right track by choosing `princess all`.

**Comment by Neato-Nick** [comment]
Gotcha, and since it's all read-based phasing, phasing the nanopore samples
with the pacbio wouldn't actually improve the phasing overall right?
Phasing the whole population together is only important for statistical
phasing tools?

**Comment by MeHelmy** [comment]
If you run each technology separately (which we discussed earlier) then "phasing the nanopore samples
with the BacBio wouldn't actually improve the phasing overall right?" will not take place, because we will use each technology separately to identify and phase the variants that were identified by this specific technology. 
"Phasing the whole population together is only important for statistical
phasing tools?" Right because it will give you the bigger picture about how these variants look like in the population.

Best,
Medhat

**Comment by MeHelmy** [comment]
Currently, Princess works on diploid genome, so I am assuming that your sample is diploid.

**Comment by Neato-Nick** [comment]
Yep I'm working with a diploid

Big thank you for this discussion, I appreciate the help! Think I've got a
clear path moving forward now.
